arch/arm/soc/nordic_nrf/nrf52/CMakeLists.txt

```diff
@@ -3,7 +3,8 @@ zephyr_compile_definitions_ifdef(CONFIG_SOC_NRF52832 NRF52832_XXAA)
 zephyr_compile_definitions_ifdef(CONFIG_SOC_NRF52840 NRF52840_XXAA)
 
 zephyr_sources(
-  mpu_regions.c
   power.c
   soc.c
   )
+
+zephyr_sources_ifdef(CONFIG_ARM_MPU_NRF52X mpu_regions.c)
```

Comment:
As far as I know we do not have non-standard MPU. So why there is CONFIG_ARM_MPU_NRF52X  option?

Reply:
This is done, in order to allow board definitions of Nordic 52 SOCs (or user projects) to build with or without support for MPU. This option depends on CPU_HAS_MPU (i.e. SOC capabilities) )and "selects" ARM_MPU (the standard MPU in ARM).